### ISSUE TITLE
[Bug🐛] Correct enum names and wire values in Serde errors

### ISSUE THREAD

**Issue by mxsm** [mxsm/rocketmq-rust#10247]
### Prerequisites

- [x] I have searched existing issues and discussions
- [x] This is a bug, not a question (use Discussions for questions)

### Environment

- OS: Any
- RocketMQ Rust: Current repository checkout
- Rust: Repository toolchain

### Bug Description

Three handwritten Serde visitors report misleading diagnostics. `ConsumeType` and `MessageModel` say they expect `TopicFilterType`, and `GroupRetryPolicyType` names an internal visitor and lists mixed-case Rust variant names even though the accepted wire values are uppercase. Valid decoding works, but invalid configuration messages point users to the wrong type or values.

### Steps to Reproduce

1. Deserialize an unknown string into `ConsumeType`, `MessageModel`, or `GroupRetryPolicyType` with `serde_json`.
2. Inspect the returned error text.
3. Observe `TopicFilterType`, `GroupRetryPolicyTypeVisitor`, or accepted values that do not match the actual wire strings.

### Expected vs Actual

**Expected:** Each error names the correct enum and lists the exact accepted wire values.

**Actual:** The diagnostics contain copied or internal names and, for two visitors, non-wire variant spellings.

### Logs/Screenshots

Not applicable. The incorrect strings are in the visitor implementations.

### Additional Context

File checklist:

- `rocketmq-protocol/src/protocol/heartbeat/consume_type.rs`: make `expecting` name `ConsumeType` and make unknown-variant diagnostics list `CONSUME_ACTIVELY`, `CONSUME_PASSIVELY`, and `CONSUME_POP`.
- `rocketmq-protocol/src/protocol/heartbeat/message_model.rs`: make `expecting` name `MessageModel` while preserving `BROADCASTING` and `CLUSTERING` as the accepted values.
- `rocketmq-protocol/src/protocol/subscription/group_retry_policy_type.rs`: make `expecting` name `GroupRetryPolicyType` and list `EXPONENTIAL` and `CUSTOMIZED`.
- Update the existing tests in those files to assert the corrected type names and accepted wire values in errors.

Acceptance criteria:

- Unknown-value errors for all three enums name the correct enum.
- The listed alternatives exactly match the strings accepted by each `visit_str` implementation.
- Serialization, valid deserialization, `Display`, and enum defaults do not change.
- Tests assert useful fragments rather than the complete `serde_json` line and column suffix.

Validation:

```text
cargo test -p rocketmq-protocol consume_type
cargo test -p rocketmq-protocol message_model
cargo test -p rocketmq-protocol group_retry_policy_type
cargo fmt -p rocketmq-protocol -- --check
cargo clippy -p rocketmq-protocol --tests --no-deps -- -D warnings
```
